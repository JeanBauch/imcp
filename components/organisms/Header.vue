<template>
	<header class="bg-primary-bordo-500 w-full rounded-b-3xl md:rounded-none absolute z-40 top-0" :class="(isHomeToBlur) ? 'md:bg-white-glass md:backdrop-blur-lg' : ''">
		<Navbar />
		<NavbarMobile />
		<NavbarDesktop />
	</header>
</template>

<script lang="ts">
import Vue from 'vue';
import Navbar from '../molecules/Navbar.vue';
import NavbarDesktop from '../molecules/NavbarDesktop.vue';
import NavbarMobile from '../molecules/NavbarMobile.vue';

export default Vue.extend({
	components: { Navbar, NavbarMobile, NavbarDesktop },
	props: {
		isHomeToBlur: {
			type: Boolean,
			default: false,
		},
	},
});
</script>
