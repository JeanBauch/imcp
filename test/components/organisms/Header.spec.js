import { mount, RouterLinkStub } from '@vue/test-utils';

import Header from '~/components/organisms/Header.vue';

describe('Components:organisms:Header', () => {
	const buildWrapper = () => {
		return {
			stubs: {
				RouterLink: RouterLinkStub,
			},
		};
	};
	describe('Snapshots', () => {
		it('should render component', () => {
			const config = buildWrapper();
			const wrapper = mount(Header, config);

			expect(wrapper.html()).toMatchSnapshot();
		});
	});
});
