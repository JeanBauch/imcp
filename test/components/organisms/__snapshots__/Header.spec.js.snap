// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components:organisms:Header Snapshots should render component 1`] = `
"<header class=\\"bg-primary-bordo-500 w-full rounded-b-3xl md:rounded-none absolute z-40 top-0\\">
  <div class=\\"container mx-auto flex items-center justify-between md:justify-center px-6 sm:px-9 py-4\\">
    <div class=\\"text-white min-h-[35px]\\"><img src=\\"logos/logo-canarinhos.svg\\" alt=\\"Logo Canarinhos\\"></div>
    <div class=\\"menu-btn\\">
      <div class=\\"menu-burguer\\"></div>
    </div>
  </div>
  <nav id=\\"navList\\" class=\\"h-0 container mx-auto items-center justify-start px-6 sm:px-9 rounded-b-3xl transition-all ease-in-out duration-700 md:hidden\\">
    <ul class=\\"flex flex-col gap-2 border-t-2 border-gray border-opacity-20 w-full pt-4 invisible opacity-0\\">
      <li class=\\"text-white hover:font-semibold\\"><a>
          Início
        </a></li>
      <li class=\\"text-white hover:font-semibold\\"><a>
          Instituto
        </a></li>
      <li class=\\"text-white hover:font-semibold\\"><a>
          Escola de Música
        </a></li>
      <li class=\\"text-white hover:font-semibold\\"><a>
          Midia
        </a></li>
      <li class=\\"text-white hover:font-semibold\\"><a>
          Noticias
        </a></li>
      <li class=\\"text-white hover:font-semibold\\"><a>
          Agenda
        </a></li>
      <li class=\\"text-white hover:font-semibold\\"><a>
          Contato
        </a></li>
      <li class=\\"text-white hover:font-semibold\\"><a>
          Transparência
        </a></li>
    </ul>
  </nav>
  <nav class=\\"hidden md:flex mx-auto\\">
    <ul class=\\"flex justify-center items-center gap-7 xl:gap-12 2xl:gap-16 border-t-2 border-b-2 border-gray border-opacity-20 w-full\\">
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Início
          </h5>
        </a></li>
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Instituto
          </h5>
        </a></li>
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Escola de Música
          </h5>
        </a></li>
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Midia
          </h5>
        </a></li>
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Noticias
          </h5>
        </a></li>
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Agenda
          </h5>
        </a></li>
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Contato
          </h5>
        </a></li>
      <li class=\\"text-white hover:text-primary-yellow-500\\"><a>
          <h5 class=\\"py-4 border-t-2 border-transparent hover:border-t-primary-yellow-500 expect-border-header\\">
            Transparência
          </h5>
        </a></li>
    </ul>
  </nav>
</header>"
`;
